libs/dps/dtm0_lib.py

```diff
@@ -0,0 +1,161 @@
+                                       num_clients=workload['num_clients'],
+                                       num_sample=workload['num_sample'],
+                                       obj_name_pref=workload['obj_name_pref'],
+                                       obj_size=workload['obj_size'],
+                                       skip_cleanup=skipcleanup,
+                                       skip_write=True,
+                                       skip_read=skipread,
+                                       validate=validate,
+                                       log_file_prefix=f"read_workload_{workload['obj_size']}mb",
+                                       end_point=S3_CFG["s3_url"],
+                                       validate_certs=S3_CFG["validate_certs"])
+                self.log.info(f"Workload: %s objects of %s with %s parallel clients ",
+                              workload['num_sample'], workload['obj_size'],
+                              workload['num_clients'])
+                self.log.info(f"Log Path {resp[1]}")
+                log_path = resp[1]
+                if s3bench.check_log_file_error(resp[1]):
+                    results.append(False)
+                    break
+                else:
+                    results.append(True)
+
+        if all(results):
+            queue.put([True, f"S3bench workload is successful. Last read log file {log_path}"])
+        else:
+            queue.put([False, f"S3bench workload for failed."
+                              f" Please read log file {log_path}"])
+
+    def process_restart(self, master_node, pod_prefix, container_prefix, process,
+                        recover_time: int = 30):
+        """
+        Restart specified Process of specific pod and container
+        :param master_node: Master node object
+        :param pod_prefix: Pod Prefix
+        :param container_prefix: Container Prefix
+        :param process: Process to be restarted.
+        :param recover_time: Wait time for process to recover
+        """
+        pod_list = master_node.get_all_pods(pod_prefix=pod_prefix)
+        pod_selected = pod_list[random.randint(0, len(pod_list) - 1)]
+        self.log.info("Pod selected for m0d process restart : %s", pod_selected)
+        container_list = master_node.get_container_of_pod(pod_name=pod_selected,
+                                                          container_prefix=container_prefix)
+        container = container_list[random.randint(0, len(container_list)-1)]
+        self.log.info("Container selected : %s", container)
+        self.log.info("Perform m0d restart")
+        resp = master_node.kill_process_in_container(pod_name=pod_selected,
+                                                     container_name=container,
+                                                     process_name=process)
+        self.log.debug("resp : %s", resp)
+        time.sleep(recover_time)
```

Comment:
so planning to use fixed wait time or we can poll with some timeout, maybe once u implement how to check if process restarted, u can give a thought for it.

Reply:
Sure. Will add check if the container and process is restarted. 
Will take care of this in subsequent PRs.